src/js/containers/search/SearchContainer.jsx

```diff
@@ -125,66 +179,29 @@ const SearchContainer = ({ history }) => {
             .catch(() => {
                 setDownloadInFlight(false);
             });
-    }, [filters, appliedFilters.filters]);
+    }, [stagedFilters, appliedFilters]);
 
     useEffect(() => {
-        const areAppliedFiltersEmpty = SearchHelper.areFiltersEqual(appliedFilters.filters, initialState);
-        if (areAppliedFiltersEmpty) {
-            // all the filters were cleared, reset to a blank hash
-            dispatch(setAppliedFilterEmptiness(true));
-            dispatch(setAppliedFilterCompletion(true));
-            history.replace('/search');
-            return;
-        }
-        else if (!inFlight) {
+        const areFiltersCleared = SearchHelper.areFiltersEqual(appliedFilters, initialState);
+        const shouldResetHash = (
```

Comment:
We do not need to be creating variables for this, a simple comment will suffice.